### PR TITLE
Search multiple words

### PR DESCRIPTION
**Pull request by sandcha**
Connected to #84 

Manage multiple words search.
First words have more importance than next ones.
Multiple matching in name has priority.
And it looks for matching in name and in description but doesn't look for word repetition in name and in description.

### CODE REVIEW

**Comment by fpagnoux** [comment]
If I type `salaire base`, I get results I should not have:

image

**Comment by fpagnoux** [comment]
Sorry I had to rewrite some of the code , but the demo is coming 😱 .

As I can't review what is not here anymore, some feedback:

- In `JS`, as in duck-typed languages in general, you are quite flexibles on types. For instance  `1 + true == 2`. So  `counter += condition ? 1 : 0` can be easily replaced by `counter += condition`.

- When you want to iterate on the items of an array, the fancy and brand new `for (const item of array) {` is a nicer than `for (const item in array) {`. The latter will iterate on the keys `i`, and you then have to get the value with `array[i]`, which is very heavy. If you can't use `ES6` on a other project, `array.forEach(...)` is still a better alternative than iterating on keys.

In general, it's more readable to avoid working with the indexes, unless you _really_ have to (and you rarely have).

- For this kind of code, automated testing is crucial. The algorithm is a little tricky, and you don't want to manually check that you still respect _all_ the spec _each time_ you change something. 

Otherwise, storing the scores as properties was really clever, and the sorting was well done 👍 .

**Comment by fpagnoux** [comment]
I'm not able to review this, as I edited the code.
@MattiSG do you have time ?
Other @sandcha or @Anna-Livia ?

File: .gitignore [hunk 1]
```diff
@@ -4,5 +4,6 @@
 /public/
 /tags
 /webpack-assets.json
+.tmp/mocha-webpack/
```
Comment (fpagnoux) [inline .gitignore]: We don't use `mocha-webpack` anymore ;)

File: tests/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+/npm-debug.log
```
Comment (fpagnoux) [inline tests/.gitignore]: Why not putting this in the root `.gitignore` ?

**Comment by sandcha** [comment]
Ok with your code changes as well.

**Comment by fpagnoux** [comment]
Ok, rebasing and merging 🚀

**Pull request merged**